src/invidious/routes/api/manifest.cr

```diff
@@ -1,6 +1,10 @@
 module Invidious::Routes::API::Manifest
   # /api/manifest/dash/id/:id
   def self.get_dash_video_id(env)
+    if !CONFIG.invidious_companion.empty?
+      return error_template(403, "This endpoint is not permitted because it is handled by Invidious companion.")
+    end
```

Comment:
What do you think here to redirect to a random invidious companion, rather than display an error message? Sure, if there are multiple companions, we might not hit the same that initially loaded the watch page, but it might make playback smoother.

Reply:
The issue is that you have to do a request to a random invidious companion for knowing its "public" URL. I had ideas to include the public URL inside the config.yml. But my poor Crystal knowledge limited me due to the new URIArrayConverter.

Something like:
```
invidious_companion:
 - http://127.0.0.1:8282:
      public_url: https://companion1.invidious.com
```

I want to avoid doing unnecessary requests to invidious companion.

`http://127.0.0.1:8282` is the internal address that invidious uses to communicate with invidious companion. but the companion could very well be on another server, and so having a config like this can exist:

```
invidious_companion:
 - http://10.0.0.2:8282:
      public_url: https://companion1.invidious.com
```

10.0.0.2 is another server and 10.0.0.0/24 is an internal network faster than the internet network. example: https://www.ovhcloud.com/en/public-cloud/private-network/

Reply:
I think that the simplest way to do it would be with an intermediate class, like that. Then the `URIArrayConverter` is not needed anymore, you can just use the simpler `URIConverter`

```
class CompanionConfig
  @[YAML::Field(converter: Preferences::URIConverter)]
  property internal_url : URI

  @[YAML::Field(converter: Preferences::URIConverter)]
  property public_url : URI
end

class Config
  # Invidious companion
  property invidious_companion : Array(CompanionConfig) = [] of CompanionConfig
end
```